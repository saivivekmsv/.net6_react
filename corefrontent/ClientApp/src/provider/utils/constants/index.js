import { mapperFormFields } from "..";
import {
  companyFormFields,
  planFormFields,
  eligibilityFormFields,
  payrollFormFields,
  censusFormFields,
  maintenanceFormFields,
  reportsFormFields,
} from "./";

export * from "./common";
export * from "./company";
export * from "./plan";
export * from "./eligibility";
export * from "./employee";
export * from "./payroll";
export * from "./maintenance";
export * from "./reports";
export * from "./enrollment";
export * from "./mapper";
export * from "./docoCenter";
export * from "./doco";
export const formFields = {
  ...companyFormFields,
  ...planFormFields,
  ...eligibilityFormFields,
  ...censusFormFields,
  ...payrollFormFields,
  ...maintenanceFormFields,
  ...reportsFormFields,
  ...mapperFormFields,
};
