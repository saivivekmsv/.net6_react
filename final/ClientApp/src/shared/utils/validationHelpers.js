export const required = (value) => {
  return [undefined, null, ""].includes(value) ? "Required" : undefined;
};

export const requiredNumber = (value) => {
  return [undefined, null, "", 0].includes(value) ? "Required" : undefined;
};
