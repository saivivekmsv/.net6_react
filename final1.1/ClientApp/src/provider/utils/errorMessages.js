export const errors = {
  companyName: {
    required: "Required",
    alreadyExist: "Company Name already exist",
  },
  companyType: {
    required: "Required",
  },
  sponsoringOrganizationId: {
    required: "Required",
    alreadyExist: "PL016 : Sponsoring Organization Name already exists",
    lengthExceeded:
      "Sponsoring Organization Name should not exceed 50 characters",
    characters:
      "Sponsoring Organization Name should not contain special characters",
  },
  masterClassificationName: {
    required: "CM048 : Required",
    alreadyExist: "CM049 : Classification Type already exists",
    length:
      "CM050 : Classification Type length should not exceed 50 characters",
    characters:
      "CM051 : Classification Type length should not contain special characters",
  },
  masterEmploymentStatus: {
    required: "CM052 : Required",
    alreadyExist: "CM053 : Employment Status already exists",
    length: "CM054 : Employment status should not exceed 50 characters",
    characters:
      "CM055 : Employment status should not contain special characters",
  },
  masterLoanTypeExists: {
    required: "Required",
    alreadyExist: "Loan Type already exists",
    length: "Loan Type length should not exceed 50 characters",
    characters: "Loan Type should not contain special characters",
  },
  einNo: {
    required: "Required",
    alreadyExist: "Tax EIN already exist",
  },
  planName: {
    required: "Required",
    alreadyExist: "Plan Name already exist",
  },
  rkPlanNumber: {
    required: "Required",
    alreadyExist: "Record Keeping Plan number already exists",
  },
  payrollFilterDateOverlap: {
    overlapError: "File received from and to date should not get overlapped",
  },
};

export const apiErrors = {
  404: {
    heading: "Unable to retrieve your data",
    reason: "Something went wrong! please try again.",
  },
  400: {
    heading: "Invalid Input",
    reason: "Please verify and re enter your details.",
  },
  500: {
    heading: "OOPS!",
    reason: "Something went wrong! please try again.",
  },
};
