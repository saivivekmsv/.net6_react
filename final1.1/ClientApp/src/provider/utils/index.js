export * from "./constants";
export * from "./helpers";
export * from "./errorMessages";
export * from "./validationHelpers";
export * from "./normaliser";
export * from "./dataTransformers";
