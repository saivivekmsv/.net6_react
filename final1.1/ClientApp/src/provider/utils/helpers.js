import React from "react";
import { Route } from "react-router-dom";
import {
  manageCompanyFormNames,
  managePlanFormNames,
  managePayrollFormNames,
} from "./";
import {
  map,
  get,
  isArray,
  memoize,
  isEmpty,
  toLower,
  upperFirst,
} from "lodash";
import { FLOW_TYPES, OPTIONS_DATA_MAPPER, planFormFields } from "./constants";
import AuthorizeRoute from "../components/api-authorization/AuthorizeRoute";

export const getCommaSeparatedValuesFromArr = (arr = []) => arr.join(", ");
export const populateManagePayrollFilterForm = (data) => {
  return {
    [managePayrollFormNames.PAYROLL_FILTER]: {
      companyId: get(data, "companyId"),
      planId: get(data, "planId"),
      fileStatus: get(data, "fileStatus"),
      fromDate: get(data, "fromDate"),
      toDate: get(data, "toDate"),
    },
  };
};
export const populateManageCompanyForm = (data) => {
  return {
    [manageCompanyFormNames.DEFINITIONS_MANAGE_COMPANY]: {
      companyName: get(data, "name"),
      companyType: get(data, "employerType"),
      sponsoringOrganizationId: get(data, "sponsoringOrganizationId"),
    },
    [manageCompanyFormNames.INCORPORATION_DETAILS_MANAGE_COMPANY]: {
      companyName: get(data, "name"),
      address1: get(data, "address1"),
      address2: get(data, "address2"),
      country: get(data, "country", "USA"),
      state: get(data, "state"),
      city: get(data, "city"),
      postalCode: get(data, "postalCode"),
      phoneNumber: get(data, "phoneNumber"),
      email: get(data, "email"),
      website: get(data, "website"),
      businessType: get(data, "businessType"),
      taxedAs: get(data, "taxedAs"),
      fisicalYearMonth: get(data, "fisicalYearMonth"),
      fisicalYearDay: get(data, "fisicalYearDay"),
      stateOfIncorporation: get(data, "stateOfIncorporation"),
      companyStartDate: get(data, "companyStartDate"),
      taxEIN: get(data, "taxEIN"),
      rkPlanNumber: get(data, "rkPlanNumber"),
      companyImage: get(data, "companyImage"),
    },
    [manageCompanyFormNames.SETTINGS_MANAGE_COMPANY]: {
      modeOfHours: get(data, "modeOfHours"),
      modeOfCompensation: get(data, "modeOfCompensation"),
      isPayrollCalenderRequire: get(data, "isPayrollCalenderRequire"),
      isFileGenerationRequire: get(data, "isFileGenerationRequire"),
      modeOfFunding: get(data, "modeOfFunding"),
    },
  };
};

export const populateManagePlanForm = (data) => {
  // return data;
  return {
    [managePlanFormNames.CREATE_PLAN]: {
      category: get(data, "category"),
      planType: get(data, "planType"),
      prototype: get(data, "prototype"),
      company: get(data, "company"),
      planName: get(data, "name"),
      planID: get(data, "id"),
      planLevel: get(data, "level"),
      AECompany: get(data, "company"),
      masterPlan: get(data, "masterPlan"),
      sponsoringOrganisation: get(data, "sponsoringOrganisation.name"),
      sponsoringOrganizationId: get(data, "sponsoringOrganisation.id"),
    },
    [managePlanFormNames.BASIC_DETAILS_MANAGE_PLAN]: {
      planName: get(data, "planName"),
      yearStartMonth: get(data, "startDate.month"),
      yearStartDate: get(data, "startDate.day"),
      yearEndMonth: get(data, "endDate.month"),
      yearEndDate: get(data, "endDate.day"),
      effectiveDate: get(data, "effectiveDate"),
      planTerminationDate: get(data, "planTerminationDate"),
      shortYearIndicator: get(data, "shortYearIndicator"),
      shortYearStartDate: get(data, "shortYearStartDate"),
      shortYearEndDate: get(data, "shortYearEndDate"),
      irsPlanNumber: get(data, "irsPlanNumber") || "",
      trsContractId: get(data, "trsContractId") || "",
      name: get(data, "planAdministrator.name"),
      mobilePhoneNumber:
        get(data, "planAdministrator.contact.mobilePhoneNumber") ||
        get(data, "planAdministrator.contact.workPhoneNumber"),
      email: get(data, "planAdministrator.contact.email"),
    },
    [managePlanFormNames.ADD_SPONSOR_MANAGE_PLAN]: {
      firstName: get(data, "firstName"),
      lastName: get(data, "lastName"),
      ssn: get(data, "ssn"),
      email: get(data, "email"),
      phoneNumber: get(data, "phoneNumber"),
    },
    [managePlanFormNames.ADD_TRUSTEE_MANAGE_PLAN]: {
      trusteeCompanyName: get(data, "trusteeCompanyName"),
      email: get(data, "email"),
      mobilePhoneNumber: get(data, "mobilePhoneNumber"),
      workPhoneNumber: get(data, "workPhoneNumber"),
      website: get(data, "website"),
      level: get(data, "level"),
      addToMaster: get(data, "addToMaster"),
    },
    [managePlanFormNames.MANAGE_RETIREMENT]: {
      earlyRetirement: get(data, "earlyRetirement"),
      earlyAgeYears: get(data, "earlyAgeYears"),
      earlyAgeMonths: get(data, "earlyAgeMonths"),
      earlyServiceYears: get(data, "earlyServiceYears"),
      earlyRetirementDateRule: get(data, "earlyRetirementDateRule"),
      normalRetirement: get(data, "normalRetirement"),
      normalAgeYears: get(data, "normalAgeYears"),
      normalAgeMonths: get(data, "normalAgeMonths"),
      normalServiceYears: get(data, "normalServiceYears"),
      normalRetirementDateRule: get(data, "normalRetirementDateRule"),
    },
    [manageCompanyFormNames.BASIC_SOURCES]: {
      sourceType: get(data, "sourceType"),
      sourceCategory: get(data, "sourceCategory"),
      sourceSubCategory: get(data, "sourceSubCategory"),
      sourceName: get(data, "sourceName"),
      recordKeepingNumber: get(data, "recordKeepingNumber"),
      effectiveStartDate: get(data, "effectiveStartDate"),
      isNewContributionAllowed: get(data, "isNewContributionAllowed"),
      isDisplayToParticipantWebsite: get(data, "isDisplayToParticipantWebsite"),
      isContributionManatory: get(data, "isContributionManatory"),
      contributionType: get(data, "contributionType"),
      limitMinimum: get(data, "limitMinimum"),
      limitMaximum: get(data, "limitMaximum"),

      maximumDollarCompensation: get(data, "maximumDollarCompensation"),
      rehireDeferralPercentage: get(data, "rehireDeferralPercentage"),
      contributionMethod: get(data, "contributionMethod"),
      allocationPercForRehires: get(data, "allocationPercForRehires"),
      otherPercForRehires: get(data, "otherPercForRehires"),
      shouldEnrollInAdiProgramme: get(data, "shouldEnrollInAdiProgramme"),
      adiApplicableTo: get(data, "adiApplicableTo"),
      periodOfIncrease: get(data, "periodOfIncrease"),
      autoDeferralIncrease: get(data, "autoDeferralIncrease"),
      maximumADI: get(data, "maximumADI"),
      subjectToAutoEnrollment: get(data, "subjectToAutoEnrollment"),
      defaultContributionForAutoEnrollment: get(
        data,
        "defaultContributionForAutoEnrollment"
      ),
      enrollmentType: "enrollmentType",
      responsibleMode: "responsibleMode",
      eligiblePaymentType: "eligiblePaymentType",
      employerContributionType: "employerContributionType",
      percentageOfCompensation: "percentageOfCompensation",
      periodForCalculation: "periodForCalculation",
      periodForMatchCalculation: "periodForMatchCalculation",
      sourceOfMatch: "sourceOfMatch",
      isSafeHarbourMatch: "isSafeHarbourMatch",
      isCalculationSuspended: "isCalculationSuspended",
      isCatchApplicable: "isCatchApplicable",
      isTrueUpApplicable: "isTrueUpApplicable",
      safeHarbourType: "safeHarbourType",
      enhancedshMatchPercentage: "enhancedshMatchPercentage",
      uptoPercentageOfDeferral: "uptoPercentageOfDeferral",
      percentageOfCalculation: "percentageOfCalculation",
      percentageOfCompensationMatched: "percentageOfCompensationMatched",
      firstTierMatchPercent: "firstTierMatchPercent",
      secondTierMatchPercent: "secondTierMatchPercent",
      thirdTierMatchPercent: "thirdTierMatchPercent",
      firstTierCompensationMatchPercent: "firstTierCompensationMatchPercent",
      secondTierCompensationMatchPercent: "secondTierCompensationMatchPercent",
      thirdTierCompensationMatchPercent: "thirdTierCompensationMatchPercent",
      employeeGroupsApplicable: "employeeGroupsApplicable",
      qmacType: "qmacType",
      qmacBefore1989: "qmacBefore1989",
      qnecBefore1989: "qnecBefore1989",
    },
    [managePlanFormNames.ADD_ELIGIBILITY]: {
      eligibilityRule: get(data, "eligibilityRuleFor"),
      eligibilityName: get(data, "name"),
      eligibilityDescription: get(data, "description"),
      exclusionType: get(data, "exclusionFromRule.exclusionType"),
      hire: get(data, "exclusionFromRule.dateOfHireType"),
      immediateEligibility: get(data, "immediateEligibility"),
      employeePopulation: get(data, "isRevaluationRequired"),
      isBreakInService: get(data, "isBreakInService"),
      applicableEligibilityRequirement: get(data, "eligibilityType"),
      yearsOfServiceDefinition: get(data, "yearsOfServiceDefinition"),
      eligibilityOption: get(data, "eligibilityOption"),
      age: get(data, "age"),
      yearsOfServiceHours: get(data, "elapsedTime.yearsOfServiceHours"),
      activeEmployee: get(data, "activeEmployee"),
      disabledEmployee: get(data, "disabledEmployee"),
      deceasedEmployee: get(data, "deceasedEmployee"),
      retiredEmployee: get(data, "retiredEmployee"),
      daily: get(data, "daily"),
      weekly: get(data, "weekly"),
      biweekly: get(data, "biweekly"),
      semiMonthly: get(data, "semiMonthly"),
      monthly: get(data, "monthly"),
      quarterly: get(data, "quarterly"),
      semiAnnually: get(data, "semiAnnually"),
      annually: get(data, "annually"),
      eligibilittyCalculationPeriod: get(
        data,
        "defaultEligibilityRule.eligibilittyCalculationPeriod"
      ),
      yearsOfServiceRequirement: get(
        data,
        "actualHours.yearsOfServiceRequirement"
      ),
      serviceCreditPeriod: get(data, "actualHours.serviceCreditPeriod"),
      consecutiveBreaksInService: get(data, "consecutiveBreaksInService"),
      breakInServiceDefinition: get(data, "breakInServiceDefinition"),
      sourceEligibilities: get(data, "sourceEligibilities"),
      hiredOnOrAfterDate: get(data, "exclusionFromRule.hiredOnOrAfterDate"),
      hiredOnOrBeforeDate: get(data, "exclusionFromRule.hiredOnOrBeforeDate"),
      // lengthOfServiceRequirement: get(
      //   data,
      //   "elapsedTime.lengthOfServiceRequirement"
      // ),
      monthsYearDropdown: get(data, "monthsYearDropdown"),
      // lengthOfService: get(data, "elapsedTime.lengthOfService"),
      otherComputationalMonths: get(data, "otherComputationalMonths"),
      hours: get(data, "hours"),
      lengthOfService: get(data, "lengthOfService"),
      period: get(data, "period"),
      eligibilityBreakInServices: get(
        data,
        "eligibilityBreakInServices",
        []
      ).map(({ breakInServiceRuleId }) => breakInServiceRuleId),
    },
  };
};

export const getPathWithParam = ({ path, pathParam, queryParam }) => {
  const isPathParamString =
    pathParam !== "" && ["number", "string"].includes(typeof pathParam);
  if (isPathParamString || pathParam) {
    const params = isArray(pathParam)
      ? pathParam.join("/").replace(`${FLOW_TYPES.ADD}/`, "")
      : pathParam;
    return `${path}/${params}${queryParam || ""}`;
  }
  return `${path}${queryParam || ""}`;
};

export const getAdvancedPathWithParam = ({ path, pathParam, queryParam }) => {
  const isPathParamString =
    pathParam !== "" && ["number", "string"].includes(typeof pathParam);
  if (isPathParamString || pathParam) {
    const params = isArray(pathParam) ? pathParam.join("/") : pathParam;
    return `${path}/${params}${queryParam || ""}`;
  }
  return `${path}${queryParam || ""}`;
};

export const getFlowBasedFormValues = (formValues, flow) => {
  if (flow === FLOW_TYPES.ADD) {
    return {};
  }
  return formValues;
};

export const tranformListToDropdownValues = (list = [], labelKey, valueKey) => {
  return map(list, (item) => ({
    label: item[labelKey],
    value: item[valueKey],
  }));
};

export const usDateFormat = (date) => {
  if (!date) {
    return null;
  }
  const jsDate = typeof date === "string" ? new Date(date) : date;
  // const dayAsStr = jsDate.toLocaleDateString();
  const dayAsStr =
    (jsDate.getMonth() > 8
      ? jsDate.getMonth() + 1
      : "0" + (jsDate.getMonth() + 1)) +
    "/" +
    (jsDate.getDate() > 9 ? jsDate.getDate() : "0" + jsDate.getDate()) +
    "/" +
    jsDate.getFullYear();
  return dayAsStr;
};

export const DateFormat = (date, symbol) => {
  if (!date) {
    return null;
  }
  const jsDate = typeof date === "string" ? new Date(date) : date;
  // const dayAsStr = jsDate.toLocaleDateString();
  const dayAsStr =
    (jsDate.getMonth() > 8
      ? jsDate.getMonth() + 1
      : "0" + (jsDate.getMonth() + 1)) +
    symbol +
    (jsDate.getDate() > 9 ? jsDate.getDate() : "0" + jsDate.getDate()) +
    symbol +
    jsDate.getFullYear();
  return dayAsStr;
};

export const getNullTableItem = (data) => {
  return !["", null, undefined].includes(data) ? (
    data
  ) : (
    <span className="null-data">NA</span>
  );
};

export const returnOnlyIfBoolean = (value) => {
  const isBoolean = typeof value === "boolean";
  if (isBoolean) {
    return value;
  }
  return undefined;
};

export const toOptionValuesFromMapper = memoize((obj) => {
  return Object.keys(obj).map((key) => ({
    label: obj[key],
    value: !isNaN(key) ? parseInt(key, 10) : key,
  }));
});

export const clearFieldValues = ({
  values = {},
  fieldsToExculde = [],
  fieldsToClear = [],
}) => {
  const fieldValues = {};
  Object.keys(values).forEach((key) => {
    if (
      (!isEmpty(fieldsToExculde) && !fieldsToExculde.includes(key)) ||
      (!isEmpty(fieldsToClear) && fieldsToClear.includes(key))
    ) {
      fieldValues[key] = null;
    } else {
      fieldValues[key] = values[key];
    }
  });
  return fieldValues;
};

export const getRandomThreeDigitValue = () => {
  return Math.floor(Math.random() * (999 - 100 + 1) + 100);
};

export const normaliseClassificationValues = (classifications = []) => {
  const normalisedClassification = classifications.map((item) => {
    return {
      ...item,
      id: item.id < 0 ? undefined : item.id,
      employeeClassificationType: parseInt(item.employeeClassificationType, 10),
      isNew: undefined,
      employeeClassificationCodes: get(
        item,
        "employeeClassificationCodes",
        []
      ).map((codeItem) => ({
        ...codeItem,
        id: codeItem.id < 0 ? undefined : codeItem.id,
        isNew: undefined,
        attributes: get(codeItem, "attributes", []).map((attrItem) => ({
          ...attrItem,
          id: attrItem.id < 0 ? undefined : attrItem.id,
          isNew: undefined,
        })),
      })),
    };
  });
  return normalisedClassification;
};

export const getFieldsToBeTouched = (fields = {}) => {
  const temp = {};
  Object.keys(fields).forEach((field) => {
    temp[field] = true;
  });
  return temp;
};

export const getSingleCharDayName = (dayNumber) => {
  const dayMapper = {
    0: "S",
    1: "M",
    2: "T",
    3: "W",
    4: "T",
    5: "F",
    6: "S",
  };
  return dayMapper[dayNumber];
};

export const ApplicationRoute =
  process.env.NODE_ENV === "development" ? Route : AuthorizeRoute;

export const getSaveMessage = (isEdit, createMessage) => {
  return isEdit || !createMessage ? "Data saved successfully" : createMessage;
};

export const getFilteredInvestments = ({
  list,
  selectedInvestmentSearch,
  selectedInvestmentType,
  selectedInvestmentStatus,
}) => {
  return list.filter((item) => {
    const isInvestmentNameMatching =
      toLower(item.name).indexOf(toLower(selectedInvestmentSearch)) !== -1;

    const isInvestmentTypeMatching = selectedInvestmentType === item.type;

    const isInvestmentStatusMatching = selectedInvestmentStatus === item.status;

    if (
      !selectedInvestmentSearch &&
      !selectedInvestmentType &&
      (isNaN(selectedInvestmentStatus) || selectedInvestmentStatus === "")
    ) {
      return true;
    }

    if (
      isInvestmentNameMatching &&
      isInvestmentTypeMatching &&
      isInvestmentStatusMatching
    ) {
      return true;
    }

    if (
      (isNaN(selectedInvestmentStatus) || selectedInvestmentStatus === "") &&
      isInvestmentNameMatching &&
      isInvestmentTypeMatching
    ) {
      return true;
    }

    if (
      !selectedInvestmentType &&
      isInvestmentNameMatching &&
      isInvestmentStatusMatching
    ) {
      return true;
    }

    if (
      !selectedInvestmentSearch &&
      isInvestmentTypeMatching &&
      isInvestmentStatusMatching
    ) {
      return true;
    }

    if (
      !selectedInvestmentType &&
      (isNaN(selectedInvestmentStatus) || selectedInvestmentStatus === "") &&
      isInvestmentNameMatching
    ) {
      return true;
    }

    if (
      !selectedInvestmentSearch &&
      (isNaN(selectedInvestmentStatus) || selectedInvestmentStatus === "") &&
      isInvestmentTypeMatching
    ) {
      return true;
    }

    if (
      !selectedInvestmentSearch &&
      !selectedInvestmentType &&
      isInvestmentStatusMatching
    ) {
      return true;
    }
    return false;
  });
};

export const getScrollParent = (node, selector) => {
  if (node == null) {
    if (selector) {
      return document.querySelector(selector);
    }
    return null;
  }

  if (node.scrollHeight > node.clientHeight + 3) {
    return node;
  } else {
    return getScrollParent(node.parentNode, selector);
  }
};

export const getPlanBasicDetailsHeader = (values) => {
  const dataArr = [];
  const fields = planFormFields[managePlanFormNames.CREATE_PLAN];
  const category = OPTIONS_DATA_MAPPER.PLAN_CATERGORY[values[fields.category]];
  const planLevel = OPTIONS_DATA_MAPPER.PLAN_LEVEL[values.level];
  const planType = OPTIONS_DATA_MAPPER.PLAN_TYPES[values[fields.planType]];
  const isMepOrPep = ["MEP", "PEP"].includes(category);
  if (category) {
    dataArr.push({
      label: "Category",
      value: category,
    });
  }

  if (["Regular"].includes(category)) {
    dataArr.push({ label: "Type", value: planType });
    dataArr.push({
      label: "Prototype",
      value: values[fields.prototype] === 0 ? null : values[fields.prototype],
    });
    dataArr.push({ label: "Company", value: values[fields.companyName] });
  }
  if (isMepOrPep && planLevel === "Master") {
    dataArr.push({ label: "Level", value: planLevel });
    dataArr.push({ label: "Type", value: planType });
    dataArr.push({
      label: "Sponsoring Org.",
      value: get(values, "sponsoringOrganisation.name", ""),
    });
  }
  if (isMepOrPep && planLevel !== "Master") {
    dataArr.push({ label: "Level", value: planLevel });
    dataArr.push({ label: "Master Plan", value: values[fields.mepPlanId] });
    dataArr.push({ label: "Company", value: values[fields.companyId] });
  }
  if (["Prototype"].includes(category)) {
    dataArr.push({ label: "Type", value: planType });
  }
  return dataArr;
};

export const getFormattedSsn = (value) => {
  let val = (value || "").replace(/\D/g, "");
  val = val.replace(/^(\d{3})(\d{1})/, "$1-$2");
  val = val.replace(/(\d)-(\d{2})(\d{1})/, "$1-$2-$3");
  val = val.replace(/(\d)-(\d)-(\d{4}).*/, "$1-$2-$3");
  return val;
};
export const getNumber = (value) => {
  return value === 0 ? value : value || undefined;
};
export const getFormattedPhone = (value) => {
  let val = (value || "").replace(/\D/g, "");
  val = val.replace(/^(\d{3})(\d{1})/, "$1-$2");
  val = val.replace(/(\d)-(\d{3})(\d{1})/, "$1-$2-$3");
  val = val.replace(/(\d)-(\d)-(\d{4}).*/, "$1-$2-$3");
  return val;
};

export const getFormattedZip = (value) => {
  let val = (value || "").replace(/\D/g, "");
  val = val.replace(/^(\d{5})(\d{1})/, "$1-$2");
  val = val.replace(/(\d)-(\d{4}).*/, "$1-$2");
  return val;
};

export const getFormattedTaxEIN = (value) => {
  let val = (value || "").replace(/\D/g, "");
  val = val.replace(/^(\d{2})(\d{1})/, "$1-$2");
  val = val.replace(/(\d)-(\d{7}).*/, "$1-$2");
  return val;
};

export const transformMasterDataInfo = (list = []) => {
  return list.map((item) => ({
    ...item,
    email: get(item, "contactDetails.email"),
    phoneNumber:
      get(item, "contactDetails.mobilePhoneNumber") ||
      get(item, "contactDetails.workPhoneNumber"),
  }));
};

export const contextSharing = {
  setContext: (contextId, context = {}) => {
    sessionStorage.setItem(contextId, JSON.stringify(context));
  },
  getContext: (contextId) => {
    return JSON.parse(sessionStorage.getItem(contextId));
  },
  clearContext: (contextId) => {
    sessionStorage.removeItem(contextId);
  },
};

export const transformToMultiselectSave = (list, key) => {
  return (list || []).map((value) => ({
    id: 0,
    [key]: value,
  }));
};

export const replaceSpaceAndParseInt = (str = "") => {
  return parseInt(str.replace(/\s/g, ""), 10);
};

export const toFlowLayoutHeader = (header, flow) => {
  if (FLOW_TYPES.EDIT === flow) {
    return (header || "").replace(/add/gi, "");
  }
  return header;
};

export const toCurrency = (num = 0) => {
  return num.toLocaleString();
};

export const scrollToErrorControl = (name) => {
  const elem = document.querySelector(`[name="${name}"]`);
  if (elem) {
    elem.scrollIntoView({ block: "center", behavior: "smooth" });
  }
};

export const ssnMasking = (ssn = "") => {
  return ssn.length == 0
    ? null
    : ssn.replaceAll("-", "").length < 9
    ? ssn
    : `XXX-XX-${ssn.slice(-4)} `;
};

export const getParamsFromQueryString = (param) => {
  const urlParams = new URLSearchParams(window.location.search);
  return (urlParams && urlParams.get(param)) || "";
};

export const stripHyphenForApi = (str = "") => {
  if (!str) {
    return "";
  }
  return str.replace(/-/g, "");
};

export function numberWithCommas(x) {
  return x.toLocaleString("en");
}

export const toMultiSelectValue = (selectedValues = [], options) => {
  if (isEmpty(selectedValues)) {
    return "";
  } else if (selectedValues.length > 1) {
    return `${selectedValues.length} Selected`;
  } else if (selectedValues.length === 1) {
    const result = options.filter((option) => option.id === selectedValues[0]);
    return result.length > 0 ? result[0].name : "";
  }
};

export const toMultiSelectValueById = (selectedValues = [], options) => {
  if (isEmpty(selectedValues)) {
    return "";
  } else if (selectedValues.length > 1) {
    return `${selectedValues.length} Selected`;
  } else if (selectedValues.length === 1) {
    const result = options.filter(
      (option) => option.value === selectedValues[0]
    );
    console.log("result", options);
    return result.length > 0 ? result[0].label : "";
  }
};

export const toSentenceCase = (str) => upperFirst(toLower(str));

export const formatDateForApiRequest = (date) => {
  let date_array = date.replaceAll("/", "-").split("-").reverse();
  return `${date_array[0]}-${date_array[2]}-${date_array[1]}`;
};

export const getMultiSelectListFromValue = (val, lastIndex) => {
  let result = [];
  for (let i = 1; i <= lastIndex; i++) {
    if ((i & val) === i) {
      result.push(i);
    }
  }
  return result;
};

export const getKeyUsingValue = (object, value) => {
  for (var key in Object.keys(object)) {
    if (object[key] === value) {
      return key;
    }
  }
  return undefined;
};
