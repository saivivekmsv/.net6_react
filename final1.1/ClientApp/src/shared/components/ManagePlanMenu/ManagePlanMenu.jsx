import React from "react";
import { get } from "lodash";
import { FLOW_TYPES, getPathWithParam, MANAGE_PLAN_ROUTES } from "../../utils";
import { Link } from "../";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faCheckCircle } from "@fortawesome/pro-light-svg-icons";

const ManagePlanMenu = ({
  menuList = [],
  selectedMenu,
  flow,
  planId,
  companyId,
  completedModules,
}) => {
  // console.log(companyId, "companyIdinMenyu");

  return (
    <ul className="manage-company-menu">
      {menuList.map((item) => {
        // console.log(item.path, "companyIdinMenyu");
        const childRoutes = get(item, "childRoutes", []);
        let className =
          selectedMenu.indexOf(item.path) !== -1 ||
          childRoutes.filter((route) => {
            return selectedMenu.indexOf(route) !== -1;
          }).length
            ? "selected"
            : "";

        const tickStatus = completedModules.includes(item.id) ? "done" : "";
        className = `${className} ${tickStatus}`;
        const disabled =
          !planId && item.path !== MANAGE_PLAN_ROUTES.CREATE_PLAN;
        className = disabled ? `${className} disabled` : className;

        if (planId && item.path === MANAGE_PLAN_ROUTES.CREATE_PLAN) {
          return null;
        }

        if (!planId && item.path === MANAGE_PLAN_ROUTES.BASIC_DETAILS) {
          return null;
        }

        if (item.path === MANAGE_PLAN_ROUTES.MANAGE_COMPENSATION) {
          return (
            <li key={item.path} className={`${className} has-tick`}>
              <FontAwesomeIcon icon={faCheckCircle} size="1x" />
              &nbsp;&nbsp;&nbsp;&nbsp;
              <Link
                disabled={disabled}
                to={getPathWithParam({
                  path: item.path,
                  pathParam: [FLOW_TYPES.EDIT, planId, companyId],
                })}
              >
                {item.label}
              </Link>
              <span className="open-rounded-corner-top" />
              <span className="open-rounded-corner-bottom" />
            </li>
          );
        }

        return (
          <li key={item.path} className={`${className} has-tick`}>
            <FontAwesomeIcon icon={faCheckCircle} size="1x" />
            &nbsp;&nbsp;&nbsp;&nbsp;
            <Link
              disabled={disabled}
              to={getPathWithParam({
                path: item.path,
                pathParam: [FLOW_TYPES.EDIT, planId],
              })}
            >
              {item.label}
            </Link>
            <span className="open-rounded-corner-top" />
            <span className="open-rounded-corner-bottom" />
          </li>
        );
      })}
    </ul>
  );
};

export default ManagePlanMenu;
