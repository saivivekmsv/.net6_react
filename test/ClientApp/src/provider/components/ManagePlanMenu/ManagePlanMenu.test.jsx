import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import ManagePlanMenu from "./ManagePlanMenu";
import { MANAGE_PLAN_MENU, MANAGE_PLAN_ROUTES } from "../../utils";

describe("ManagePlanMenu", () => {
  const props = {
    menuList: MANAGE_PLAN_MENU,
    selectedMenu: MANAGE_PLAN_ROUTES.BASIC_DETAILS,
  };
  test("to match snapshot", () => {
    const { baseElement } = render(
      <Router>
        <ManagePlanMenu {...props} />
      </Router>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
