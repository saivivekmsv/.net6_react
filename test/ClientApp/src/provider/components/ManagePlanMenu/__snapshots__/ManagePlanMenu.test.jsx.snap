// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagePlanMenu to match snapshot 1`] = `
<body>
  <div>
    <ul
      class="manage-company-menu"
    >
      <li
        class="selected"
      >
        <a
          href="/manage-plan/basic-details"
        >
          Basic Details
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/sponsor"
        >
          Sponsor
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/trustee"
        >
          Trustee
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/custodian"
        >
          Custodian
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/advisor"
        >
          Advisor
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/complaince"
        >
          Complaince
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/manage-sources"
        >
          Sources
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/eligibility"
        >
          Eligibility
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/allocation"
        >
          Allocation
        </a>
      </li>
      <li
        class=""
      >
        <a
          href="/manage-plan/vesting"
        >
          Vesting
        </a>
      </li>
    </ul>
  </div>
</body>
`;
